src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -781,13 +781,28 @@ void AppHost::_HandleSummon(const winrt::Windows::Foundation::IInspectable& /*se
     {
         if (_LazyLoadDesktopManager())
         {
-            GUID currentlyActiveDesktop{ 0 };
-            if (VirtualDesktopUtils::GetCurrentVirtualDesktopId(&currentlyActiveDesktop))
+            // First thing - make sure that we're not on the current desktop. If
+            // we are, then don't call MoveWindowToDesktop. This is to mitigate
+            // MSFT:33035972
+            BOOL onCurrentDesktop{ false };
+            if (SUCCEEDED(_desktopManager->IsWindowOnCurrentVirtualDesktop(_window->GetHandle(), &onCurrentDesktop)) && onCurrentDesktop)

```

Comment:
`FAILED` doesn't exist? Or could we do `!SUCCEEDED`?

I was taught that...
```cpp
if (x)
   // do nothing
else
   // do something
```

should always be simplified to...
```cpp
if (!x)
   // do something
```

but otherwise, meh.

Reply:
This is more complicated because of the conditional. I'd just leave it.